web/src/app/services/ServiceDeleteDialog.js

```diff
@@ -61,34 +63,42 @@ export default function ServiceDeleteDialog({ serviceID, onClose }) {
     variables: { id: serviceID },
   })
   const input = [{ type: 'service', id: serviceID }]
-  const dispatch = useDispatch()
   const [deleteService, { loading, error }] = useMutation(mutation, {
     variables: { input },
-    onCompleted: () => dispatch(push('/services')),
   })
 
-  if (dataLoading) return <Spinner />
+  const epID = _.get(data, 'service.ep.id')
```

Comment:
is `_.get` necessary for this one since there is no fallback if it's undefined? As opposed to simply `const epID = data.service.ep.id`

Reply:
The get is an alternative to `data && data.service && data.service.ep && data.service.ep.id` since `data.service` can exist with`ep` being null, and `data` without `service`.

No fallback is OK in this case, because without ep the form can't get submitted/is disabled, however we still want to continue rendering.

Reply:
Do you thinking pulling in the lodash package here is worth it over the `&&`s?

Reply:
I think it's clearer to read and easier maintain. Pulling in lodash doesn't cost anything other than the line of the import statement, since we're already using lodash in the application. Webpack puts it in a single global/shared "module", so the code is always downloaded and only "executed" once for the entire page on first import.

Might be worth a quick discussion if we should always use it going forward though (At least for >1 level deep). It also supports the use of a `default` value which is handy elsewhere.

 `foo && foo.bar && foo.bar.baz && foo.bar.baz.bin || _default`
vs
`_.get(foo, 'bar.baz.bin', _default)`

Esp. since the first one only works because `&&` takes precedence over `||`, otherwise you'd need to put parens. in there too.